flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/SortMergeResultPartition.java

```diff
@@ -492,10 +490,13 @@ public int getNumberOfQueuedBuffers(int targetSubpartition) {
     }
 
     private int[] getRandomSubpartitionOrder(int numSubpartitions) {
-        List<Integer> list =
-                IntStream.range(0, numSubpartitions).boxed().collect(Collectors.toList());
-        Collections.shuffle(list);
-        return list.stream().mapToInt(Integer::intValue).toArray();
+        int[] order = new int[numSubpartitions];
+        Random random = new Random();
+        int shift = random.nextInt(numSubpartitions);
```

Comment:
If `shift` is 0, the returned array may not be changed. 

In addition, this patch may cause the actual traversal order not to change, but only the starting position to be modified. Is this expected?

Reply:
Yes, it is expected. The motivation is for better sequential disk IO when there is no enough resources to run the downstream consumer tasks in parallel.